### ISSUE TITLE
CS8774 should not warn when object already known to be non-null

### ISSUE THREAD

**Issue by lonix1** [dotnet/roslyn#75576]
**Version Used**: 
.NET 8.0.110

**Steps to Reproduce**:
```cs
Potato? _potato;

[MemberNotNull(nameof(_potato))]
public async Task FetchPotatoAsync()
{
    if (_potato != null) return;

    await Task.Delay(100);       // WARNING: Member '_potato' must have a non-null value when exiting. [CS8774]

    _potato = new Potato();
}
```

**Diagnostic Id**:

CS8774: Member '_potato' must have a non-null value when exiting.

**Expected Behavior**:
Should not warn.

**Actual Behavior**:
Does warn.

This is discussed in https://github.com/dotnet/csharplang/discussions/5657. However this variation of that problem can be fixed - in a lazy-loaded scenario, the object is already known to be non-null.

**Comment by 333fred** [comment]
I don't see how this is any different than the standard scenario here, which cannot be fixed without a language change. When the await is hit, _potato is null. And that will violate the contract of the attribute.
